bdb/temptable.c

```diff
@@ -620,6 +666,17 @@ int bdb_temp_table_create_pool_wrapper(void **tblp, void *bdb_state_arg)
     return bdberr;
 }
 
+int bdb_temp_table_notify_pool_wrapper(void **tblp, void *bdb_state_arg)
+{
+    bdb_state_type *bdb_state = (bdb_state_type *)bdb_state_arg;
+    int rc1 = recover_deadlock_simple(bdb_state);
```

Comment:
We definitely do not want to unconditionally recover deadlock every time we request a temp table - it is rare that we run out of temptables, hence most of the time we'd be doing unnecessary work here. Maybe only do this when the pool is both full (# items reaches upper limit) and exhausted (no idle items) -- ie when comdb2_objpool_available() is false.

Reply:
Joe pointed out that we get in here exactly when the pool is both full and exhausted and the caller chooses to wait. So this is sane.